### DISCUSSION TITLE
GitHub Search code badge

### DISCUSSION THREAD

**Discussion by ccoVeille** [badges/shields#10684] [Ideas] [answered]
I'm opening an idea, because maybe it exists.
If not, I will open an issue.

I would like to get badge that would provide the number of results for a GitHub search query

I often need to report the number of results that a GitHub query would match

Something like:

> [len([]byte(...](https://github.com/search?q=language%3Ago+%22len%28%5B%5Dbyte%28%22&type=code): 6.5k 
> [len(string(](https://github.com/search?q=language%3Ago+%22len%28string%28%22&type=code): 5.4k 
> [len(json.RawMessage(...](https://github.com/search?q=language%3Ago+%22len%28json.RawMessage%28%22&type=code): 4

I had used [here](https://github.com/dominikh/go-tools/issues/760#issuecomment-2480989858)

I also need sometimes things like [that](https://github.com/editorconfig-checker/editorconfig-checker/pull/363#issuecomment-2480530118)

> The directive was used [twice](https://github.com/search?q=%22editorconfig-checker-disable-next-line%22+-is%3Afork+-org%3Aeditorconfig-checker&type=code) in 3 months

Some other people also have [this needs](https://github.com/go-simpler/musttag/issues/106#issuecomment-2480767298_)

> For example, if I use more precise search queries:
> - [query1](https://github.com/search?q=path%3A%2F.golangci*+%22musttag%3A%22+%22goccy%2Fgo-json%22+-path%3A*reference*+&type=code): only 4 projects (by mainly only one user) has defined a configuration for `github.com/goccy/go-json`
> -  [query2](https://github.com/search?q=path%3A%2F.golangci*+%22musttag%3A%22+%22github.com%2Fjson-iterator%2Fgo%22+-path%3A*reference*+&type=code): only 1 projects has defined a configuration for `github.com/json-iterator/go`.
>
> And if you use those queries ([A](https://github.com/search?q=%28path%3A%2Fgo.mod++AND+%22github.com%2Fjson-iterator%2Fgo%22%29&type=code), > [B](https://github.com/search?q=%28path%3A%2Fgo.mod++AND+%22github.com%2Fgoccy%2Fgo-json%22%29&type=code)) you will see that most of the references a related to indirect dependencies.
 

So is there something equivalent to this via shields.io?

A badge that will search on GitHub, provide the results and will open the link to GitHub the exact search page (by default)
 
Here I'm searching for code, but we could consider something about repositories/issues/....

Thanks

**Comment by chris48s** [comment]
We already have the GitHub Search Hit Counter https://shields.io/badges/git-hub-search-hit-counter badge which gives you the number of search hits within a repo
e.g: https://img.shields.io/github/search/badges/shields/BaseService
..and if you want, you could link that to a search result page e.g:

```
[![](https://img.shields.io/github/search/badges/shields/BaseService)](https://github.com/search?q=repo%3Abadges%2Fshields%20BaseService&type=code)
```

[![](https://img.shields.io/github/search/badges/shields/BaseService)](https://github.com/search?q=repo%3Abadges%2Fshields%20BaseService&type=code)

**Reply by ccoVeille** [reply]
Thanks for replying me.

I also found this badge. I should have talked about it.

Unfortunately, it doesn't work for my need.

It is limited to a repository. My need is to search globally on GitHub. So not restricted to a repository

**Reply by chris48s** [reply]
OK, so fundamentally all that boils down to: a version of https://shields.io/badges/git-hub-search-hit-counter which is not repo scoped.

**Reply by ccoVeille** [reply]
Exactly

**Reply by ccoVeille** [reply]
based on @chris48s messages that somehow confirmed it was not present and an existing one could be reused, I created

I created : 
- https://github.com/badges/shields/issues/10686
 
and opened

- https://github.com/badges/shields/pull/10687

**Reply by ccoVeille** [reply, marked as answer]
The badge is now available https://shields.io/badges/git-hub-code-search-count

**Comment by chris48s** [comment]
There's also the GitHub Issues Custom Search badge https://shields.io/badges/git-hub-issue-custom-search
With that one, you can use any query searching for issues/PRs (see https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests ) e.g: `repo:badges/shields is:closed label:bug author:app/sentry-io` to get a count matching that query

https://img.shields.io/github/issues-search?query=repo%3Abadges%2Fshields%20is%3Aclosed%20label%3Abug%20author%3Aapp%2Fsentry-io

..and then again, you could make it a link when you embed e.g:

```
[![](https://img.shields.io/github/issues-search?query=repo%3Abadges%2Fshields%20is%3Aclosed%20label%3Abug%20author%3Aapp%2Fsentry-io
)](https://github.com/badges/shields/issues?q=is%3Aclosed+label%3Abug+author%3Aapp%2Fsentry-io)
```

[![](https://img.shields.io/github/issues-search?query=repo%3Abadges%2Fshields%20is%3Aclosed%20label%3Abug%20author%3Aapp%2Fsentry-io
)](https://github.com/badges/shields/issues?q=is%3Aclosed+label%3Abug+author%3Aapp%2Fsentry-io)

or whatever

**Reply by ccoVeille** [reply]
This one is about looking for issues with a search. It's pretty close to my needs, but I want to look for code, not issues.

So my need might be close to a copy paste of this badge

So the code would be a mix of

https://github.com/badges/shields/blob/master/services%2Fgithub%2Fgithub-issues-search.service.js

and

https://github.com/badges/shields/blob/master/services%2Fgithub%2Fgithub-search.service.js
